src/contracts/pods/EigenPodManager.sol

```diff
@@ -243,6 +533,119 @@ contract EigenPodManager is Initializable, OwnableUpgradeable, Pausable, IEigenP
+    function _addShares(address podOwner, uint256 shareAmount) internal {
+        require(podOwner != address(0), "EigenPodManager.restakeBeaconChainETH: podOwner cannot be zero address");
+        require(shareAmount > 0, "EigenPodManager.restakeBeaconChainETH: amount must be greater than zero");
+
+        podOwnerShares[podOwner] += shareAmount;
+
+        // if the podOwner has delegated shares, record an increase in their delegated shares
+        if (!isInUndelegationLimbo(podOwner)) {
+            delegationManager.increaseDelegatedShares(podOwner, beaconChainETHStrategy, shareAmount);
+        }
+    }
+
+    // @notice Reduces the `podOwner`'s shares by `shareAmount` and performs a call to the DelegationManager to ensure delegated shares are also tracked correctly
+    function _removeShares(address podOwner, uint256 shareAmount) internal {
+        require(podOwner != address(0), "EigenPodManager._removeShares: depositor cannot be zero address");
+        require(shareAmount != 0, "EigenPodManager._removeShares: shareAmount should not be zero!");
+
+        uint256 currentPodOwnerShares = podOwnerShares[podOwner];
+        require(shareAmount <= currentPodOwnerShares, "EigenPodManager._removeShares: shareAmount too high");
+
+        unchecked {
+            currentPodOwnerShares = currentPodOwnerShares - shareAmount;
+        }
+
+        podOwnerShares[podOwner] = currentPodOwnerShares;
+
+        // if the podOwner has delegated shares, record a decrease in their delegated shares
+        if (!isInUndelegationLimbo(podOwner)) {
+            uint256[] memory shareAmounts = new uint256[](1);
+            shareAmounts[0] = shareAmount;
+            IStrategy[] memory strategies = new IStrategy[](1);
+            strategies[0] = beaconChainETHStrategy;
+            delegationManager.decreaseDelegatedShares(podOwner, strategies, shareAmounts);
+        }
+    }
+
+    // @notice Changes the `podOwner`'s shares by `sharesDelta` and performs a call to the DelegationManager to ensure delegated shares are also tracked correctly
+    function _updateSharesToReflectBeaconChainETHBalance(address podOwner, int256 sharesDelta) internal {
+        if (sharesDelta < 0) {
+            // if change in shares is negative, remove the shares
+            _removeShares(podOwner, uint256(-sharesDelta));
+        } else {
+            // if change in shares is positive, add the shares
+            _addShares(podOwner, uint256(sharesDelta));
+        }      
+    }
+
+    /**
+     * @notice This function is called to decrement withdrawableRestakedExecutionLayerGwei when a validator queues a withdrawal.
+     * @param amountWei is the amount of ETH in wei to decrement withdrawableRestakedExecutionLayerGwei by
+     */
+    function _decrementWithdrawableRestakedExecutionLayerGwei(address podOwner, uint256 amountWei) 
+        internal onlyWhenNotPaused(PAUSED_WITHDRAW_RESTAKED_ETH) 
```

Comment:
why can't the pausing flag be `PAUSED_WITHDRAWALS`?

Reply:
this is already paused when `PAUSED_WITHDRAWALS` is set. I'm going to remove this modifier since it is redundant.